### PR TITLE
Cache zmq

### PR DESCRIPTION
**Pull request by jdfreder**
In theory this should really speed up the tests....

### CODE REVIEW

**Comment by jdfreder** [comment]
Also caches node_modules...  Which we may not want to do?

**Comment by jdfreder** [comment]
Actually caching node_modules should be fine as long as we don't cache package.json...

**Comment by rgbkrk** [comment]
I've seen caching `node_modules` be a problem at times, though I don't have anything concrete beyond that.

File: travis_before_install [hunk 1]
```diff
@@ -8,20 +8,28 @@ export LD_RUN_PATH=$LDHACK/lib
...
-'[ -z "$SODIUM" ] || tar xzvf libsodium-$SODIUM.tar.gz'
-'[ -z "$SODIUM" ] || cd libsodium-$SODIUM'
-'[ -z "$SODIUM" ] || ./autogen.sh'
-'[ -z "$SODIUM" ] || ./configure --prefix=$LDHACK'
-'[ -z "$SODIUM" ] || make'
-'[ -z "$SODIUM" ] || make install'
-'[ -z "$SODIUM" ] || cd ..'
-'[ -z "$SODIUM" ] || export LIBS=-lsodium && export sodium_CFLAGS=$CFLAGS && export sodium_LIBS=$LDFLAGS'
+
+if [ ! -d "zeromq-$ZMQ" ]; then
```
Comment (rgbkrk) [inline travis_before_install:12]: 👍

**Comment by jdfreder** [comment]
4 min 46 sec -> 1 min 46 sec

That's 37% of the uncached time!

**Comment by rgbkrk** [comment]
:boom:

**Pull request merged**